sdks/go/pkg/beam/core/runtime/graphx/translate.go

```diff
@@ -445,6 +446,9 @@ func (m *marshaller) addMultiEdge(edge NamedEdge) ([]string, error) {
 			payload.RestrictionCoderId = coderId
 			m.requirements[URNRequiresSplittableDoFn] = true
 		}
+		if _, ok := edge.Edge.DoFn.ProcessElementFn().BundleFinalization(); ok {
+			m.requirements[URNRequiresBundleFinalization] = true
```

Comment:
Tiniest note:
Out of an abundance of caution, I paranoid check how we handle requirements on serialization.

Could you add a sort.Strings() call on the returned slice of requirement urns from getRequirements? It will help avoid spurious diffs in pipeline protos at some time later. 

https://github.com/apache/beam/blob/35490f37bf17510b7516591c191045b77a567d21/sdks/go/pkg/beam/core/runtime/graphx/translate.go#L218

Reply:
Sure - it should be super cheap anyways since there aren't many requirements.